### DISCUSSION TITLE
A route question

### DISCUSSION THREAD

**Discussion by zhllucky** [vercel/next.js#72821] [Help] [answered]
### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/quirky-driscoll-sxkjss?file=%2Fnext.config.mjs

### To Reproduce

In the latest version of my Next.js project, I use (with-nav-bar) as a folder in my directory structure. Within it, I have a dashboard folder that contains page.tsx, which generates a /dashboard page. At the same level as (with-nav-bar), I also have another dashboard folder without page.tsx, but with secondary pages. This setup generates routes like /dashboard/a. However, after building the project, I find that if I directly access /dashboard, it redirects to /dashboard/, resulting in a 404 error. Why does this happen, and how can it be resolved?

### Current vs. Expected behavior

![image](https://github.com/user-attachments/assets/ae57b211-5a8e-4b94-91cd-617e4e5767cc)

### Provide environment information

```bash
no info
```

### Which area(s) are affected? (Select all that apply)

create-next-app

### Which stage(s) are affected? (Select all that apply)

next build (local)

**Comment by samcx** [comment]
@zhllucky I don't think the reproduction is laid out as described in your **To Reproduce**! If you can update your reproduction, we will be able to take a look.

**Reply by zhllucky** [reply]
It's not easy to describe this issue on the codesandbox. If you want to reproduce it completely, you should first execute yarn build, and then access it through nginx service startup or http server startup in the out file. This will result in redirection 404. If accessed through npx serve@latest Out is also normal, which is quite strange. You can try reproducing it.
![Screenity video - Nov 15, 2024](https://github.com/user-attachments/assets/2731dcae-0fdc-4866-bd7f-ed730025325d)

**Comment by zhllucky** [comment]
I still think this is a bug and it's very hidden

**Reply by icyJoseph** [reply]
Is this a trailing slash issue? Like, your nginx config is using trailing slash, but your next.js config is not?

**Reply by zhllucky** [reply]
No, you can try following my process and use httpserve to start it

**Reply by icyJoseph** [reply]
Could you make a stackblitz reproduction demo? Just type next.new into your browser address bar. Codesandbox is not a very good way to show these.

**Reply by zhllucky** [reply]
https://stackblitz.com/edit/stackblitz-starters-tv5fu8?file=README.md   
you can use this, you need run 'npm run build', and then cd out folder，and then run npx http-server，and view /dashboard

**Reply by icyJoseph** [reply, marked as answer]
Yeah, it looks like an issue with the server rather than Next.js

I used `serve` https://www.npmjs.com/package/serve, and it works. 

I also see that:

```console
~/projects/stackblitz-starters-tv5fu8 1m 11s
❯ ls ./out/dashboard
game.html  game.txt   

~/projects/stackblitz-starters-tv5fu8
❯ ls ./out/
404.html        dashboard/      dashboard.txt   index.html      next.svg        
_next/          dashboard.html  favicon.ico     index.txt       vercel.svg      

~/projects/stackblitz-starters-tv5fu8
```

`dashboard.html` exists, so it should be reachable, you need to look into the options when serving with `http-server` or which ever other static server. The only way I could get `http-server` to work correctly, was by telling Next.js to use trailing-slashes, but that's not ideal either. I just don't have time right now to look into the `http-server` documentation options, to see which flag you need to pass to get it to do what you need. Alternatively, use `serve`, https://www.npmjs.com/package/serve ~
